python/cortex.py

```diff
@@ -27,16 +29,353 @@
 MENTAL_COMMAND_BRAIN_MAP_ID         =   17
 MENTAL_COMMAND_TRAINING_THRESHOLD   =   18
 SET_MENTAL_COMMAND_ACTIVE_ACTION_ID =   19
+HAS_ACCESS_RIGHT_ID                 =   20
+GET_CURRENT_PROFILE_ID              =   21
+GET_CORTEX_INFO_ID                  =   22
+UPDATE_MARKER_REQUEST_ID            =   23
+
+#define error_code
+ERR_PROFILE_ACCESS_DENIED = -32046
+
+# define warning code
+CORTEX_STOP_ALL_STREAMS = 0
+CORTEX_CLOSE_SESSION = 1
+USER_LOGIN = 2
+USER_LOGOUT = 3
+ACCESS_RIGHT_GRANTED = 9
+ACCESS_RIGHT_REJECTED = 10
+PROFILE_LOADED = 13
+PROFILE_UNLOADED = 14
+CORTEX_AUTO_UNLOAD_PROFILE = 15
+EULA_ACCEPTED = 17
+DISKSPACE_LOW = 19
+DISKSPACE_CRITICAL = 20
+HEADSET_CANNOT_CONNECT_TIMEOUT = 102
+HEADSET_DISCONNECTED_TIMEOUT = 103
+HEADSET_CONNECTED = 104
+HEADSET_CANNOT_WORK_WITH_BTLE = 112
+HEADSET_CANNOT_CONNECT_DISABLE_MOTION = 113
 
 
 class Cortex(Dispatcher):
+
+    _events_ = ['socket_disconnected','create_session_done', 'query_profile_done', 'load_unload_profile_done', 
+                'save_profile_done', 'get_mc_active_action_done','mc_brainmap_done', 'mc_action_sensitivity_done', 
+                'mc_training_threshold_done', 'create_record_done', 'stop_record_done','warn_cortex_close_session', 
+                'inject_marker_done', 'update_marker_done', 'export_record_done', 'new_data_labels', 
+                'new_com_data', 'new_fe_data', 'new_eeg_data', 'new_mot_data', 'new_dev_data', 
+                'new_met_data', 'new_pow_data', 'new_sys_data']
     def __init__(self, user, debug_mode=False):
-        url = "wss://localhost:6868"
-        self.ws = websocket.create_connection(url,
-                                            sslopt={"cert_reqs": ssl.CERT_NONE})
-        self.user = user
+        
+        self.session_id = ''
+        self.headset_id = ''
         self.debug = debug_mode
 
+        if ('client_id' not in user) or (' ' in user['client_id']):
+            raise ValueError('Invalid client_id')
```

Comment:
I prefer an action-oriented message like "Empty client_id. Please fill in client_id before running the example"

Reply:
Even better to make client_id a required init param for Cortex class

Reply:
yes, thank . I will update